src/modules/generateRandomNumber.js

```diff
@@ -6,8 +6,21 @@
  *
  * @return {number} A random 4-digit number
  */
+
 function generateRandomNumber() {
-  /* Write your code here */
+  const digits = new Set();
+
+  const firstDigit = Math.floor(1 + Math.random() * 9);
+
+  digits.add(firstDigit);
+
+  while (digits.size < 4) {
+    const randomDigit = Math.floor(Math.random() * 10);
+
+    digits.add(randomDigit);
+  }
+
+  return Number(Array.from(digits).join(''));
```

Comment:
The current implementation may not guarantee a 4-digit number due to the unordered nature of sets. Consider using an array to maintain the order of digits and ensure the first digit is always non-zero.